ui/packages/tidb-dashboard-lib/src/apps/SQLAdvisor/component/IndexInsightList.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useContext, useState, useRef } from 'react'
+
+import IndexInsightTable, { useSQLTunedListGet } from './IndexInsightTable'
+
+import { Space, Button, Typography, notification, Alert, Modal } from 'antd'
+import { Card, Toolbar } from '@lib/components'
+import { SQLAdvisorContext } from '../context'
+import dayjs from 'dayjs'
+
+const ONE_DAY = 24 * 60 * 60 // unit: second
+
+interface IndexInsightListProps {
+  onHandleDeactivate?: () => void
+  isDeactivating?: boolean
+}
+
+const IndexInsightList = ({
+  onHandleDeactivate,
+  isDeactivating
+}: IndexInsightListProps) => {
+  const ctx = useContext(SQLAdvisorContext)
+  const [showAlert, setShowAlert] = useState<boolean>(false)
+  const [noTaskRunning, setNoTaskRunning] = useState<boolean>(true)
+  const [showDeactivateModal, setShowDeactivateModal] = useState<boolean>(false)
+
+  const { sqlTunedList, refreshSQLTunedList, loading } = useSQLTunedListGet()
+
+  const taskRunningStatusGet = useRef(() => {
+    return ctx?.ds
+      .tuningTaskStatusGet()
+      .then((data) => {
+        setNoTaskRunning(data)
+        return data
+      })
+      .catch((e) => console.log(e))
+  })
+
+  const timer = useRef(0)
+  const startCheckTaskStatusLoop = useRef(() => {
+    clearTimeout(timer.current)
+    timer.current = window.setTimeout(async () => {
+      const _noTaskRunning = (await taskRunningStatusGet.current()) as boolean
+      if (_noTaskRunning) {
+        refreshSQLTunedList()
+        return
+      }
+      startCheckTaskStatusLoop.current()
+    }, 1000 * 60)
+  })
+
+  useEffect(() => {
+    const checkStatus = async () => {
+      const _noTaskRunning = (await taskRunningStatusGet.current()) as boolean
+      if (!_noTaskRunning) {
+        startCheckTaskStatusLoop.current()
+      }
+    }
+    checkStatus()
+  }, [])
+
+  useEffect(() => {
+    const checkSQLValidation = async () => {
+      await ctx?.ds
+        .sqlValidationGet?.()
+        .then((res) => {
+          setShowAlert(!res)
+        })
+        .catch((e) => console.log(e))
+    }
+
+    checkSQLValidation()
+  }, [ctx])
```

Comment:
How about changing it to the same as `taskRunningStatusGet` and combining the effect contents?

```suggestion
const checkSQLValidation = useRef(() => {
     return ctx?.ds
        .sqlValidationGet?.()
        .then((res) => {
          setShowAlert(!res)
        })
    })
    
  useEffect(() => {
    const checkStatus = async () => {
      const _noTaskRunning = (await taskRunningStatusGet.current()) as boolean
      if (!_noTaskRunning) {
        startCheckTaskStatusLoop.current()
      }
    }
    checkStatus()
    checkSQLValidation.current()
  }, [])
```